src/java/org/apache/cassandra/cql3/ConstraintFunctionExpression.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.cassandra.db.TypeSizes;
+import org.apache.cassandra.io.IVersionedAsymmetricSerializer;
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.schema.TableMetadata;
+
+public class ConstraintFunctionExpression
+{
+    public final ConstraintFunction constraintFunction;
+    public final ColumnIdentifier columnName;
+
+    public static final Serializer serializer = new Serializer();
+
+    public static final Map<String, ConstraintFunction> supportedConstraintFunctions = Map.of(
+        LengthConstraint.class.getName(), new LengthConstraint()
+    );
```

Comment:
The check for the supported constraint functions is done at `org.apache.cassandra.cql3.FunctionColumnConstraint.Raw` already. The map is redundant. 

In fact, I think this class `ConstraintFunctionExpression` is an unnecessary abstraction. You should achieve the same with `FunctionColumnConstraint` and `ColumnConstraint`. Please think about it.